src/header.rs

```diff
@@ -174,7 +173,7 @@ impl<T: ?Sized> From<Arc<HeaderSlice<(), T>>> for Arc<T> {
         );
 
         // Safety: `HeaderSlice<(), T>` and `T` has the same layout
-        unsafe { Arc::from_raw_inner(Arc::into_raw_inner(this) as _) }
```

Comment:
not a fan of this change, is there a reason?

Reply:
I was having RA errors about casting a thin pointer to a fat pointer, but it seems like rustc is fine with it? I'll revert it.